frontend/mulighetsrommet-veileder-flate/src/components/tabs/innsikt/BarChart.tsx

```diff
@@ -1,35 +1,82 @@
+import { Datapunkt } from './Datapunkt';
+
+const SISTE_AAR = 5;
 
 type Status = 'Arbeidstaker m. ytelse/oppf' | 'Kun arbeidstaker' | 'Registrert hos Nav' | 'Ukjent';
 
-function isOfStatusType(value: any): value is Status {
+function isOfStatusType(value: string): value is Status {
   return ['Arbeidstaker m. ytelse/oppf', 'Kun arbeidstaker', 'Registrert hos Nav', 'Ukjent'].includes(value);
 }
 
-function csvObjectArrayTilDatapunktArray(array: any[]): Datapunkt[] {
+function csvObjectArrayTilDatapunktArray(array: StatistikkFraCsvFil[]): Datapunkt[] {
   return array
-    .sort((item1, item2) => item2['Antall Måneder'] - item1['Antall Måneder'])
+    .sort((a, b) => parseInt(b['Antall Måneder']) - parseInt(a['Antall Måneder']))
     .map(item => {
       return {
-        tiltakstype: item['Tiltakstype'],
-        'Arbeidstaker m. ytelse/oppf': item['Arbeidstaker m. ytelse/oppf'],
-        'Kun arbeidstaker': item['Kun arbeidstaker'],
-        'Registrert hos Nav': item['Registrert hos Nav'],
-        Ukjent: item['Ukjent'],
+        År: item['År'],
+        tiltakstype: item.Tiltakstype,
+        'Arbeidstaker m. ytelse/oppf': parseFloat(replaceCommaWithPeriod(item['Arbeidstaker m. ytelse/oppf'])),
+        'Kun arbeidstaker': parseFloat(replaceCommaWithPeriod(item['Kun arbeidstaker'])),
+        'Registrert hos Nav': parseFloat(replaceCommaWithPeriod(item['Registrert hos Nav'])),
+        Ukjent: parseFloat(replaceCommaWithPeriod(item.Ukjent)),
         antallManeder: item['Antall Måneder'] + ' mnd',
       };
     });
 }
 
+function gjennomsnittForDatapunkterForSisteAar(
+  datapunkter: Datapunkt[],
+  antallAarTilbakeITid: number,
+  tiltakstype: string
+): Datapunkt[] {
+  const punkter = datapunkter.reverse().slice(0, antallAarTilbakeITid);
+
+  const punkterKronologisk = [...punkter.reverse()];
+  const aar = `${punkterKronologisk[0].År} - ${punkterKronologisk[punkterKronologisk.length - 1].År}`;
+
+  return punkter.reduce<Datapunkt[]>(
+    (all, next, _, { length }) => {
+      all[0]['Arbeidstaker m. ytelse/oppf'] += next['Arbeidstaker m. ytelse/oppf'] / length;
+      all[0]['Kun arbeidstaker'] += next['Kun arbeidstaker'] / length;
+      all[0]['Registrert hos Nav'] += next['Registrert hos Nav'] / length;
+      all[0].Ukjent += next.Ukjent / length;
+      all[0].tiltakstype = next.tiltakstype;
+      all[0].antallManeder = next.antallManeder;
+      return [...all];
+    },
+    [
+      {
+        'Arbeidstaker m. ytelse/oppf': 0,
+        'Kun arbeidstaker': 0,
+        'Registrert hos Nav': 0,
+        Ukjent: 0,
+        antallManeder: '',
+        tiltakstype,
+        År: aar,
+      },
+    ]
+  );
+}
+
+/**
+ * Erstatter komma med punktum siden tall fra norsk Excel eksporterer tall med komma istedenfor punktum.
+ * parseFloat derimot, forventer punktum i flyttall.
+ * @param value verdi med komma i seg
+ * @returns verdi med punktum som har erstattet komma
+ */
+function replaceCommaWithPeriod(value: string): string {
```

Comment:
Er dette potensielt mulig å gjøre som del av beskrivelsen i seksjonen om `Opplasting av statistikkfil fra datavarehuset` i README i stedet for i kode?

Reply:
Du tenker å gjøre erstatningen i Excel istedenfor i kode?

Reply:
Ja, i tilfelle det var en enkel innstilling man kunne benyttet seg av under eksporten f.eks., men ikke noe stress om det ikke er mulig. Så bare potensiale for å trimme litt kode :smile:

Reply:
Ja, tenkte på det, men Excel er Excel  og det er en liten ekspedisjon og endre formatering av komma vs punktum. Jeg lar det stå slik det er nå 😅